### PR TITLE
Signal Templates

### PR DESCRIPTION
**Pull request by jrs65**
Use like this:
```python
from fitstack import signal

base_path = "/project/rpp-chime/chime/stacking/sims/analysis_stacks_comp/psbeam/psbeam/"

# Load all the stack files (which have a specific naming scheme) and return
# an object to combine them
# The first argument is a pattern that should match all the relevant compderiv
# mode directories, and no more.
#
# The second argument is a scaling prefactor, for these templates use 1e3
# to get the templates in uJy, and 1e-3 to get them in Jy.
#
# The final argument renames some of the internal parameters.
# I would recommend sticking with the one below so we use then name "M_10" to control the shot noise
st = signal.SignalTemplate.load_from_stackfiles(
    base_path + "dataweight_compderiv*/post/datamask/QSO/NGC/",
    factor=1e3,
    aliases=dict(shotnoise="M_10"),
)

# You can get the set of all param names you must supply with
st.params

# The frequencies the template is defined at is available at
st.freq

# To get a template you call the `.signal` method with all the required params.
# Note these are the alpha_X scaling params and so the fiducial value is usually 1.
# You must supply them all, and they are kwonly arguments
# The output is an array of [pol, freq]
template = st.signal(omega=1, b_HI=1, b_g=1, NL=1, FoGh=1, FoGg=1, M_10=1)


```

### CODE REVIEW

File: fitstack/models.py [hunk 1]
```diff
@@ -417,3 +418,126 @@ def model(self, theta, freq=None, transfer=None):
...
+        offset = theta[0]
+
+        model_init = self._signal_template.signal(*param_dict)[[0, 3]]
+
+        model = utils.shift_and_convolve(freq, model_init, offset=offset, transfer=None)
+
+        return model
+
+
+# TODO: this would probably be cleaner if we could pass parameters through to the model
```
Comment (ssiegelx) [inline fitstack/models.py:542]: Yeah I think what I would probably do is make the `base_path` and `pattern` keyword arguments and then create a `model_kwargs` parameter to the `run_mcmc` method that is passed to the `Model` constructor.  Then you would just provide the appropriate path for your desired tracer and field in the config file.

Similarly there is a `prior_spec` parameter that is currently passed to the `Model` constructor and will override the defaults set in `_prior_spec` of the class, which will allow us to set different priors for the galaxy bias for the different tracers but use the same underlying class.

Comment (jrs65) [reply fitstack/models.py:542]: Sounds good. I didn't want to introduce a whole other bunch of changes in this first iteration, but I agree that it makes sense to make a `model_kwargs` parameter to pass through.

Thanks for the tip on the `prior_spec` I hadn't really noticed to know what that was going to do.

Testing it to see if it works now.... 🤞

**Comment by ssiegelx** [comment]
This looks good.  Unless you have any additional updates, I will go ahead and merge this.  I have made a few changes on top of this that I'll issue as a separate PR.

**Comment by jrs65** [comment]
Yeah, no other changes from me.

I found another issue which I've just avoided for now (by commenting it all out), that if I fix some of the parameters then the later stages of generating the results container fail as some of the sampler routines only return the parameters that are varying (mcmc.py L353 onwards I think). But I don't have a good fix at the moment.

**Comment by ssiegelx** [comment]
Ah okay, yeah I added the capability to fix parameters, but don't think I ever tested it.  I'll look into this.

**Pull request merged**